### PR TITLE
Render decorator documentation

### PR DESCRIPTION
**Pull request by aliabid94**
Added documentation for `@gr.render`

### CODE REVIEW

**Review by aliabd — APPROVED** [review]
Looks good from website side. Will create the docs page in the editable docs PR.

File: guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+$code_render_split_simple
+$demo_render_split_simple
+
+See how we can now create a variable number of Textboxes using our custom logic - in this case, a simple `for` loop. The `@gr.render` decorator enables this with the following steps:
+
+1. Create a function and attach the @gr.render decorator to it.
+2. Add the input components to the `inputs=` argument of @gr.render, and create a corresponding argument in your function for each component. This function will automatically re-run on any change to a component.
+3. Add all components inside the function that you want to render based on the inputs.
+
+Now whenever the inputs change, the funciton re-runs, and replaces the components created from the previous funciton run with the latest run. Pretty straightforward! Let's add a little more complexity to this app:
```
Comment (freddyaboulton) [inline guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md:20]: typo: `funciton`

File: guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+## Dynamic Event Listeners
+
+If you're creating components, you probably want to attach event listeners to them as well. Let's take a look at an example that takes in a variable number of Textbox as input, and merges all the text into a single box.
+
+$code_render_merge_simple
+$demo_render_merge_simple
+
+Let's take a look at what's happening here:
+
+1) The state variable `text_count` is keeping track of the number of Textboxes to create. By increasing it via the Add button, we trigger re-renders as `text_count` is an input to the render decorator.
```
Comment (freddyaboulton) [inline guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md:37]: This second sentence is a hard to read. Maybe something like

```
By clicking on the Add button, we increase `text_count` which triggers the render decorator
``

**Review by freddyaboulton — COMMENTED** [review]
Nice - just noticed some typos

File: guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md (hunk 2 above)
Comment (aliabd) [reply guides/04_building-with-blocks/04_dynamic-apps-with-render-decorator.md:37]: i'll fix in website pr

**Pull request merged**